addons/mail/controllers/common.py

```diff
@@ -0,0 +1,55 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http
+from odoo.http import request
+
+
+class MailControllerCommon(http.Controller):
```

Comment:
I don't think it needs to inherit from controller, as it's not using any cls.method from controller itself.

This way we avoid the diamond problem (not a problem now but it could be in the future).

Other controllers can inherit both this class and `http.Controller` by themselves.

Reply:
As discussed, the overrides won't work properly if we don't define the common class as a controller class.